demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/extra_data/ExtraDataAdapter.java

```diff
@@ -0,0 +1,58 @@
+package com.novoda.downloadmanager.demo.extended.extra_data;
+
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.novoda.downloadmanager.demo.R;
+import com.novoda.notils.caster.Views;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExtraDataAdapter extends BaseAdapter {
+    private final List<Download> downloads;
+
+    public ExtraDataAdapter() {
+        this.downloads = new ArrayList<>();
+    }
+
+    @Override
+    public int getCount() {
+        return downloads.size();
+    }
+
+    @Override
+    public Download getItem(int position) {
+        return downloads.get(position);
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return position;
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        View view = View.inflate(parent.getContext(), R.layout.list_item_extra_data_download, null);
+
+        final Download download = getItem(position);
+        TextView titleTextView = Views.findById(view, R.id.download_title_text);
```

Comment:
I see you https://github.com/novoda/download-manager/pull/79#discussion_r34242502 :alien: